### PR TITLE
refactor(experimental): group Instruction errors

### PR DESCRIPTION
**Pull request by lorisleiva**
Creates an instruction error group with its own code range.

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2251?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2252** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2252?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2251** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2251?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2250** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2250?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2249** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2249?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/errors/src/codes.ts [hunk 1]
```diff
@@ -82,6 +79,12 @@ export const SOLANA_ERROR__KEYS_INVALID_PRIVATE_KEY_BYTE_LENGTH = 3704000 as con
 export const SOLANA_ERROR__KEYS_INVALID_SIGNATURE_BYTE_LENGTH = 3704001 as const;
 export const SOLANA_ERROR__KEYS_SIGNATURE_STRING_LENGTH_OUT_OF_RANGE = 3704002 as const;
 
+// Instruction-related errors.
+// Reserve error codes in the range [4128000-4128999].
+export const SOLANA_ERROR__INSTRUCTION_EXPECTED_TO_HAVE_ACCOUNTS = 4128000 as const;
+export const SOLANA_ERROR__INSTRUCTION_EXPECTED_TO_HAVE_DATA = 4128001 as const;
+export const SOLANA_ERROR__INSTRUCTION_PROGRAM_ID_MISMATCH = 4128002 as const;
+
```
Comment (buffalojoec) [inline packages/errors/src/codes.ts:87]: Just curious if you have a method for choosing these ranges?

**Review by buffalojoec — APPROVED** [review]

File: packages/errors/src/codes.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/errors/src/codes.ts:87]: Haha I almost asked the same question to @steveluscher before starting this stack.

I just asked my brain to come up with a random number within the range I wanted the range to be in. Maybe Steve's got a more fancy answer though. 😄

Comment (steveluscher) [reply packages/errors/src/codes.ts:87]: Just don't mess with my numbering. 😚

**Review by steveluscher — APPROVED** [review]
Added via Giphy

**Comment by steveluscher** [comment]
### Merge activity

* **Mar 5, 10:18 AM PST**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2251).

**Pull request merged**